examples/objective-c/route_guide/ViewControllers.m

```diff
@@ -244,46 +238,46 @@ - (void)execRequest {
     return;
   }
 
-  GRPCStreamingProtoCall *call = [_service recordRouteWithResponseHandler:self
-                                                              callOptions:nil];
+  void (^handler)(RTGRouteSummary *response, NSError *error) =
+      ^(RTGRouteSummary *response, NSError *error) {
+        if (response) {
+          NSString *str = [NSString
+              stringWithFormat:@"%@\nFinished trip with %i points\nPassed %i features\n"
+                                "Travelled %i meters\nIt took %i seconds",
+                               self.outputLabel.text, response.pointCount, response.featureCount,
+                               response.distance, response.elapsedTime];
+          self.outputLabel.text = str;
+          NSLog(@"Finished trip with %i points", response.pointCount);
+          NSLog(@"Passed %i features", response.featureCount);
+          NSLog(@"Travelled %i meters", response.distance);
+          NSLog(@"It took %i seconds", response.elapsedTime);
+        } else {
+          NSString *str =
+              [NSString stringWithFormat:@"%@\nRPC error: %@", self.outputLabel.text, error];
+          self.outputLabel.text = str;
+          NSLog(@"RPC error: %@", error);
+        }
+      };
+
+  GRPCStreamingProtoCall *call =
+      [_service recordRouteWithResponseHandler:[[GRPCUnaryResponseHandler alloc]
```

Comment:
Why are we using `GRPCUnaryResponseHandler` for a streaming call?

Reply:
The name of the class is confusing. It's request streaming but response is unary.

Reply:
This is example code that might confuse users. Can we change the name to make it more clear?

Reply:
I don't think we can change the name of the class since it's already published. I'll make additional documentation to clarify it.